File: frontend/src/app/models/company.request.model.ts
export type Address = {
    line1: string;
    city: string;
    state: string | null;
    postal_code: string | null;
    country: string;
};

export type Company = {
    company_id: string;
    legal_name: string;
    industry: string | null;
    email: string | null;
    phone: string | null;
    country: string;
    currency: string | null;
    logo_url: string | null;
    role: string;
    status: string;
    capital_social: number | null;
    director_name: string | null;
    start_date: string | null;
    activity_description: string | null;
    addresses: Address[] | null;
};

export type CompanyUpdateForm = {
    legal_name: string;
    capital_social: string;
    director_name: string;
    start_date: string;
    industry: string;
    currency: string;
    activity_description: string;
    email: string;
    phone_prefix: string;
    phone: string;
    street: string;
    city: string;
    state: string;
    state_code: string;
    postal_code: string;
    country: string;
};

// ─── Erreurs formulaire édition ───────────────────────────────────────────────
export type CompanyFormErrors = {
    legal_name: string;
    capital_social: string;
    director_name: string;
    industry: string;
    currency: string;
    email: string;
    phone: string;
    street: string;
    city: string;
    state: string;
    country: string;
    postal_code: string;
};

// ─── Types create-enterprise (étape 1) ───────────────────────────────────────
export type CreateEnterpriseStepOneData = {
    legal_name: string;
    capital_social: string;
    director_name: string;
    start_date: string;
    industry: string;
    activity_description: string;
};

export type CreateEnterpriseStepOneErrors = {
    legal_name: string;
    director_name: string;
    industry: string;
    capital_social: string;
};

// ─── Types create-enterprise (étape 2) ───────────────────────────────────────
export type CreateEnterpriseStepTwoData = {
    email: string;
    phone: string;
    country: string;
    currency: string;
    phone_prefix: string;
    street: string;
    city: string;
    state: string;
    state_code: string;
    postal_code: string;
};

export type CreateEnterpriseStepTwoErrors = {
    email: string;
    postal_code: string;
    phone: string;
    street: string;
    country: string;
    city: string;
    state: string;
};


// ─── Valeurs initiales partagées ─────────────────────────────────────────────
export const emptyCompanyFormErrors: CompanyFormErrors = {
    legal_name: '', capital_social: '', director_name: '', industry: '',
    currency: '', email: '', phone: '', street: '', city: '',
    state: '', country: '', postal_code: '',
};

export const emptyCreateEnterpriseStepOneErrors: CreateEnterpriseStepOneErrors = {
    legal_name: '', director_name: '', industry: '', capital_social: '',
};

export const emptyCreateEnterpriseStepTwoErrors: CreateEnterpriseStepTwoErrors = {
    email: '', postal_code: '', phone: '',
    street: '', country: '', city: '', state: '',
};

File: backend/core-service/app/models/company_request_model.py
from typing import Optional
from pydantic import BaseModel


class AddressData(BaseModel):
    line1:       str
    city:        str
    country:     str
    line2:       Optional[str] = None
    state:       Optional[str] = None
    postal_code: Optional[str] = None


class CreateCompanyRequest(BaseModel):
    legal_name:           str
    capital_social:       float
    director_name:        str
    start_date:           str
    industry:             str
    email:                str
    phone:                str
    country:              str
    address:              AddressData
    activity_description: Optional[str] = None
    currency:             Optional[str] = "TND"


class UpdateCompanyRequest(BaseModel):
    legal_name:           Optional[str]       = None
    capital_social:       Optional[float]     = None
    director_name:        Optional[str]       = None
    start_date:           Optional[str]       = None
    industry:             Optional[str]       = None
    activity_description: Optional[str]       = None
    email:                Optional[str]       = None
    phone:                Optional[str]       = None
    country:              Optional[str]       = None
    currency:             Optional[str]       = None
    address:              Optional[AddressData] = None


class UpdateProfileRequest(BaseModel):
    full_name: Optional[str] = None
    phone:     Optional[str] = None
    email:     Optional[str] = None